HW1/P4/P4_bfs.py

```diff
@@ -0,0 +1,20 @@
+# Perform breadth first search starting at src_name in an adjacency list
+# rdd.  Returns an rdd which maps every node connected to src_name to its
+# distance from src_name.
+def bfs(adj_list_rdd, src_name, sc):
+    accum = sc.accumulator(0)
+    adj_list_rdd.cache()
+    distances = sc.parallelize([(src_name, 0)])
+    i = 0
+    curr = 0
+    c = adj_list_rdd.count()
+    last = -1
+    while accum.value != last:
```

Comment:
Maybe I'm misunderstanding this, but this seems like a strange approach. You're checking whether the accumulator value is equal to `last` and if it's not, you set `last` to the value, set the value to 0, and then set the accumulator value equal to the number of distances you got. Does the loop ever break when `last` is not 0? It looks to me as though you're always doing one extra iteration - it might be cleaner just to reset the accumulator on each iteration and then break the loop if it's 0 at the start of any one iteration.